src/cljs/main/broadfcui/page/library/library_page.cljs

```diff
@@ -191,52 +192,44 @@
+       [:span {} (first suggestion-parts) [:strong {} highlight] (last suggestion-parts)])
+     suggestion)])
 
-(react/defc- SearchSection
-  {:get-filters
-   (fn [{:keys [props]}]
-     (utils/map-keys name (:facet-filters props)))
-   :render
-   (fn [{:keys [props this]}]
-     [:div {:style {:paddingBottom "calc(16px - 0.9rem)"}} ;; cancel the padding on the hr and match the outer padding
-      [comps/AutocompleteFilter
-       {:ref "text-filter"
-        :on-filter (:on-filter props)
-        :typeahead-events ["typeahead:select"]
-        :width "100%"
-        :field-attributes {:defaultValue (:search-text props)
-                           :placeholder "Search"}
-        :facet-filters (:facet-filters props)
-        :bloodhoundInfo {:url (str (config/api-url-root) "/api/library/suggest")
-                         :transform (fn [response]
-                                      (clj->js
-                                       (mapv (partial hash-map :value) (aget response "results"))))
-                         :cache false
-                         :prepare (fn [query settings]
-                                    (clj->js
-                                     (assoc (js->clj settings)
-                                       :headers {:Authorization (str "Bearer " (utils/get-access-token))}
-                                       :type "POST"
-                                       :contentType "application/json; charset=UTF-8"
-                                       :data (utils/->json-string
-                                              {:searchString query
-                                               :filters (this :get-filters)
-                                               :from 0
-                                               :size 10}))))}
-        :typeaheadDisplay (fn [result]
-                            ;; underlying typeahead library uses the result of this function
-                            ;; via $input.val(x), which is safe from xss. So we explicitly
-                            ;; do not want to encode anything here.
-                            (aget result "value" "suggestion"))
-        :typeaheadSuggestionTemplate (fn [result]
-                                       (let [suggestion (aget result "value" "suggestion")
-                                             highlight (aget result "value" "highlight")
-                                             display (highlight-suggestion suggestion highlight)]
-                                         (str "<div style='textOverflow: ellipsis; overflow: hidden; font-size: smaller;'>" display "</div>")))}]])})
+(defn- create-search-section [{:keys [search-text facet-filters on-input on-filter]}]
+  [:div {:style {:paddingBottom "calc(16px - 0.9rem)"}} ;; cancel the padding on the hr and match the outer padding
+   [Autosuggest
+    {:value search-text
+     :inputProps {:placeholder "Search"}
+     :get-suggestions (fn [query callback]
```

Comment:
Can we do the `:url` thing here?

Reply:
no, because of the `POST` and the data object.